core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/AiServiceMethodImplementationSupport.java

```diff
@@ -145,6 +145,7 @@ public Object implement(Input input) {
 
     private static Object doImplement(AiServiceMethodCreateInfo methodCreateInfo, Object[] methodArgs,
             QuarkusAiServiceContext context, Audit audit) {
+        boolean isRunningOnWorkerThread = !Context.isOnEventLoopThread();
         Object memoryId = memoryId(methodCreateInfo, methodArgs, context.chatMemoryProvider != null);
         Optional<SystemMessage> systemMessage = prepareSystemMessage(methodCreateInfo, methodArgs,
                 context.hasChatMemory() ? context.chatMemory(memoryId).messages() : Collections.emptyList());
```

Comment:
@cescoffier I built a version of quarkus-langchain4j from your branch and integrated it into my project. However, I am still encountering the same exception:

```
java.lang.IllegalStateException: The current thread cannot be blocked: vert.x-eventloop-thread-3
        at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:30)
        at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
        at io.quarkus.redis.runtime.datasource.BlockingStringCommandsImpl.get(BlockingStringCommandsImpl.java:41)
        at br.com.fredericci.memory.CustomChatInMemoryStore.getMessages(CustomChatInMemoryStore.java:32)
        at br.com.fredericci.memory.CustomChatInMemoryStore_ClientProxy.getMessages(Unknown Source)
        at dev.langchain4j.memory.chat.MessageWindowChatMemory.messages(MessageWindowChatMemory.java:84)
        at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.doImplement(AiServiceMethodImplementationSupport.java:151)
        at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.implement(AiServiceMethodImplementationSupport.java:131)
        at br.com.fredericci.ChatBotAssistant$$QuarkusImpl.getAnswer(Unknown Source)
        at br.com.fredericci.ChatBotAssistant$$QuarkusImpl_ClientProxy.getAnswer(Unknown Source)
The issue seems to be linked with context.chatMemory(memoryId).messages() which relies on blocking operation 

```

as you can see in the stacktrace.